### PR TITLE
Fix/compatibility sdk issues refactor

### PR DESCRIPTION
**Pull request by frederik-encord**
Use label row structure to get the correct hashes and simplify migration code.

### CODE REVIEW

File: src/encord_active/lib/project/project_file_structure.py [hunk 1]
```diff
@@ -1,18 +1,88 @@
...
+    # older versions of Encord Active are handled with versions greater than 0.1.52.
+    label_row_meta = json.loads(pfs.label_row_meta.read_text(encoding="utf-8"))
+    with PrismaConnection() as conn:
+        fill_label_rows = conn.labelrow.count() == 0
+        fill_data_units = conn.dataunit.count() == 0
+        if not fill_label_rows or fill_data_units:
+            return
+
+        with conn.batch_() as batcher:
+            for label_row in pfs.iter_labels():
```
Comment (eloy-encord) [inline src/encord_active/lib/project/project_file_structure.py:32]: The plan is to deprecate `iter_labels()`, but I guess it's fine here and we will remove the compatibility code at the same time as that method.

Comment (eloy-encord) [reply src/encord_active/lib/project/project_file_structure.py:32]: same for `iter_data_unit()`.

File: src/encord_active/lib/project/project_file_structure.py [hunk 2]
```diff
@@ -1,18 +1,88 @@
...
+                                data_title=data_units[data_unit.hash]["data_title"],
+                                frame=data_unit.frame,
+                                location=data_unit.path.as_posix(),
+                                lr_data_hash=lr_data_hash,
+                            )
+                        )
+            batcher.commit()
+
+
+def test():
```
Comment (eloy-encord) [inline src/encord_active/lib/project/project_file_structure.py:66]: should we delete it?

File: src/encord_active/lib/project/project_file_structure.py [hunk 3]
```diff
@@ -101,30 +173,22 @@ def label_row_structure(self, label_hash: str) -> LabelRowStructure:
...
-        to_include = {"label_row": True} if include_label_row else {}
+        to_include = DataUnitInclude(label_row=True) if include_label_row else None
         with PrismaConnection() as conn:
-            # add backwards compatibility code. Remove when Encord Active version is >= 0.1.60.
-            fill_missing_tables()
-            # end backwards compatibility code.
-            if pattern is None:
-                return iter(conn.dataunit.find_many(include=to_include))
+            _fill_missing_tables(self)
+            if where is None:
```
Comment (eloy-encord) [inline src/encord_active/lib/project/project_file_structure.py:182]: [nitpick 💅 ]
`find_many()`'s `where` param is `Optional` too, so we can skip the if/else and always send the `where` value to `find_many()`.

File: src/encord_active/lib/project/project_file_structure.py [hunk 4]
```diff
@@ -1,18 +1,88 @@
+from __future__ import annotations
+
 import json
+import re
```
Comment (eloy-encord) [inline src/encord_active/lib/project/project_file_structure.py:4]: remove line to solve pylint complaint 🦺

File: src/encord_active/lib/file_structure/base.py [hunk 5]
```diff
@@ -70,7 +69,7 @@ def project_meta(self) -> Path:
 
     @abstractmethod
     def data_units(
-        self, pattern: Optional[list[DataUnitLike]] = None, include_label_row: bool = False
+        self, where: Optional[list[DataUnitWhereInput]] = None, include_label_row: bool = False
     ) -> Iterator[models.DataUnit]:
```
Comment (eloy-encord) [inline src/encord_active/lib/file_structure/base.py:73]: The output should be list as in `ProjectFileStructure`. The same for `BaseProjectFileStructure.label_rows()`.

File: src/encord_active/lib/project/project_file_structure.py (hunk 1 above)
Comment (frederik-encord) [reply src/encord_active/lib/project/project_file_structure.py:32]: Well, the entire thing is based on using that to transform it into the new structure. I think that they will be removed together these things.

File: src/encord_active/lib/project/project_file_structure.py (hunk 2 above)
Comment (frederik-encord) [reply src/encord_active/lib/project/project_file_structure.py:66]: haha yea sorry!

File: src/encord_active/lib/project/project_file_structure.py (hunk 1 above)
Comment (eloy-encord) [reply src/encord_active/lib/project/project_file_structure.py:31]: yes, please, let's nuke them in a few versions 😄

File: src/encord_active/lib/project/project_file_structure.py [hunk 6]
```diff
@@ -1,18 +1,83 @@
...
-    path: Path
+# To be deprecated when Encord Active version is >= 0.1.60.
+def _fill_missing_tables(pfs: ProjectFileStructure):
+    # Adds the content missing from the data units table when projects with
+    # older versions of Encord Active are handled with versions greater than 0.1.52.
+    label_row_meta = json.loads(pfs.label_row_meta.read_text(encoding="utf-8"))
+    with PrismaConnection() as conn:
+        fill_label_rows = conn.labelrow.count() == 0
+        fill_data_units = conn.dataunit.count() == 0
+        if not fill_label_rows or fill_data_units:
```
[the remark this answers is not in the document]
Comment (frederik-encord) [reply src/encord_active/lib/project/project_file_structure.py:27]: Damn! Good catch... fixing

File: src/encord_active/lib/file_structure/base.py [hunk 7]
```diff
@@ -1,10 +1,9 @@
 from abc import abstractmethod
 from pathlib import Path
-from typing import Iterator, Optional, Union
+from typing import Iterator, List, Optional, Union
```
Comment (eloy-encord) [inline src/encord_active/lib/file_structure/base.py:3]:
```suggestion
from typing import List, Optional, Union
```

Comment (eloy-encord) [reply src/encord_active/lib/file_structure/base.py:3]: for pylint complaint

File: src/encord_active/lib/project/project_file_structure.py [hunk 8]
```diff
@@ -1,15 +1,68 @@
...
+
+                if fill_data_units:
+                    data_units = label_row_dict["data_units"]
+                    for data_unit in label_row.iter_data_unit():
+                        du = data_units[data_unit.hash]
+                        batcher.dataunit.create(
+                            DataUnitCreateInput(
+                                data_hash=data_unit.hash,
+                                data_title=du["data_title"],
+                                frame=int(du["sequence_number"]),
```
Comment (eloy-encord) [inline src/encord_active/lib/project/project_file_structure.py:58]:
```suggestion
                                frame=int(du["data_sequence"]),
```

**Pull request merged**